tsl/src/bgw_policy/compression_api.c

```diff
@@ -276,6 +295,20 @@ policy_compression_add(PG_FUNCTION_ARGS)
 							CONFIG_KEY_COMPRESS_AFTER,
 							format_type_be(compress_after_type))));
 	}
+	/* If this is a compression policy for a cagg, verify that
+	 * compress_after > refresh_start of cagg policy. We do not want
+	 * to compress regions that can eb refreshed by the cagg policy.
+	 */
+	if (is_cagg && !policy_refresh_cagg_refresh_start_lt(hypertable->fd.id,
```

Comment:
What about changing the CAGG policy? Shouldn't there be a comparison against `compress_after` there as well for the same reason?

Reply:
Yes, the supported apis are to add/remove cagg policies.